### PR TITLE
LibJS: Small follow up parser fixes

### PR DESCRIPTION
**Pull request by davidot**
Found some small mistakes/ things I forgot in my parser PR.
And follow up to correctly parse escaped keywords where needed and fail where we should fail

(cc @trflynn89 for the escaped keyword changes)

### CODE REVIEW

**Comment by davidot** [comment]
I think the CI failures are unrelated to the change?

File: Userland/Libraries/LibJS/Token.h [hunk 1]
```diff
@@ -26,6 +26,9 @@ constexpr const u32 PARAGRAPH_SEPARATOR { 0x2029 };
 // U+00A0 NO BREAK SPACE
 constexpr const u32 NO_BREAK_SPACE { 0x00A0 };
 
+// U+FEFF ZERO WIDTH NO-BREAK SPACE
+constexpr const u32 ZERO_WIDTH_NO_BREAK_SPACE { 0xFEFF };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Token.h:30]: Please add this in the second commit instead where it's actually being used

File: Userland/Libraries/LibJS/Lexer.cpp [hunk 2]
```diff
@@ -376,7 +376,7 @@ Optional<u32> Lexer::is_unicode_escape(size_t& identifier_length) const
 {
     GenericLexer lexer(source().substring_view(m_position - 1));
 
-    if (auto code_point_or_error = lexer.consume_escaped_code_point(); !code_point_or_error.is_error()) {
+    if (auto code_point_or_error = lexer.consume_escaped_code_point(false); !code_point_or_error.is_error()) {
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Lexer.cpp:379]: What's the reason not to combine surrogate pairs?

Comment (davidot) [reply Userland/Libraries/LibJS/Lexer.cpp:379]: It's what the spec says, don't know why it is supposed to happen in some places but not others 🤷.
For reference in IdentifierName https://tc39.es/ecma262/#prod-IdentifierName we can use a UnicodeEscapeSequence https://tc39.es/ecma262/#prod-UnicodeEscapeSequence which can only be `\uXXXX` 4 hex digits or `\u{XXXXXX}` some hex digits.

Whereas ~string~ regex has https://tc39.es/ecma262/#prod-RegExpUnicodeEscapeSequence RegExpUnicodeEscapeSequence which does allow for combining surrogate pairs.

String is a little harder but has to do with source text https://tc39.es/ecma262/#sec-static-semantics-sv I think?

All in all a comment would be good here so I will add that 👍

**Comment by davidot** [comment]
Fixed the rebasing and added spec comments.

Let me know if this form of spec comment is good, kind of shoved the link in there.

**Pull request merged**